### PR TITLE
Add Precision, Recall, MeanIoU metrics.

### PR DESCRIPTION
**Pull request by pavithrasv**
### Summary
Add Precision, Recall, MeanIoU metrics.

### Related Issues
n/a

### PR Overview
- `top_k` argument in Precision/Recall metrics is supported only for TF backend.
- MeanIoU metric is supported only for TF backend

- [y] This PR requires new unit tests [y/n] (make sure tests are included)
- [n] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [y] This PR is backwards compatible [y/n]
- [n] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

File: keras/metrics.py [hunk 1]
```diff
@@ -1681,6 +1911,8 @@ def clone_metrics(metrics):
 mape = MAPE = mean_absolute_percentage_error
 msle = MSLE = mean_squared_logarithmic_error
 cosine = cosine_similarity = cosine_proximity
+if K.backend() == 'tensorflow':
+    MeanIoU = K.MeanIoU
```
Comment (fchollet) [inline keras/metrics.py]: Does that actually work with the Keras `compile`/`fit`/`evaluate` methods? If I recall correctly, there were changes made between tf.keras and external Keras (e.g. the `update_state` method returning a list).

File: keras/metrics.py [hunk 2]
```diff
@@ -1681,6 +1911,8 @@ def clone_metrics(metrics):
 mape = MAPE = mean_absolute_percentage_error
 msle = MSLE = mean_squared_logarithmic_error
 cosine = cosine_similarity = cosine_proximity
+if K.backend() == 'tensorflow':
```
Comment (fchollet) [inline keras/metrics.py]: I would suggest doing the following:

```python
BaseMeanIoU = object
if K.backend() == 'tensorflow':
  import tensorflow as tf
  if tf.__version__ >= ...:
      BaseMeanIoU = tf.keras.metrics.MeanIoU
  
class MeanIoU(BaseMeanIoU):
  def __init__(self, ...):
      if K.backend() != 'tensorflow':
         raise RuntimeError(...)
      if BaseMeanIoU is object:
         raise RuntimeError(...)

    # then override what needs to be overriden to work with this version of Keras
```

File: keras/backend/tensorflow_backend.py [hunk 3]
```diff
@@ -3501,6 +3503,24 @@ def in_top_k(predictions, targets, k):
                           k=k)
 
 
+def top_k(input, k=1, sorted=True, name=None):
```
Comment (fchollet) [inline keras/backend/tensorflow_backend.py]: Avoid adding new symbols to the backend (since they aren't present in tf.keras) and instead do an inline import of tf after checking `K.backend()`.

File: keras/metrics.py (hunk 1 above)
Comment (fchollet) [reply keras/metrics.py]: Related: please add a metric test that does, roughly, for every `M` among the object metrics:

```
model = Sequential([Dense(2, input_shape=(3,)])
model.compile('rmsprop', metrics=[M()])
model.fit(x, y)
loss_val, metric_val = model.evaluate(x, y)
```
No need to test correctness (which is tested separately).

Importantly this test does not need to exclude Theano and CNTK (so it should be in a separate file). Theano / CNTK should only be excluded specifically for MeanIoU.

Comment (pavithrasv) [reply keras/metrics.py]: Yes it works with Keras compile/fit/evaluate but i have made the changes as suggested before for raising the error messages and also added a test file.

File: keras/metrics.py (hunk 2 above)
Comment (pavithrasv) [reply keras/metrics.py]: Updated.

File: keras/backend/tensorflow_backend.py (hunk 3 above)
Comment (pavithrasv) [reply keras/backend/tensorflow_backend.py]: Removed these symbols.

**Comment by pavithrasv** [comment]
Thank you for the review!
